geode-assembly/src/integrationTest/java/org/apache/geode/tools/pulse/PulseSecurityConfigOauthProfileTest.java

```diff
@@ -0,0 +1,83 @@
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.tools.pulse;
+
+import static org.apache.geode.test.junit.rules.HttpResponseAssert.assertResponse;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.util.Properties;
+
+import org.apache.http.HttpResponse;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import org.apache.geode.examples.SimpleSecurityManager;
+import org.apache.geode.test.junit.categories.PulseTest;
+import org.apache.geode.test.junit.rules.GeodeHttpClientRule;
+import org.apache.geode.test.junit.rules.LocatorStarterRule;
+
+@Category({PulseTest.class})
+public class PulseSecurityConfigOauthProfileTest {
+  // this test just makes sure the property file in the locator's working dir
+  // gets properly read and used in the oauth security configuration
+
+  @ClassRule
+  public static LocatorStarterRule locator =
+      new LocatorStarterRule().withHttpService()
+          .withSecurityManager(SimpleSecurityManager.class)
+          .withProperty("security-auth-token-enabled-components", "pulse");
+
+  private static File pulsePropertyFile;
+
+  @BeforeClass
+  public static void setup() throws Exception {
+    // copy the pulse.properties to the locator's working dir. Pulse will use the locator's working
+    // dir as classpath to search for this property file
+    pulsePropertyFile = new File(locator.getWorkingDir(), "pulse.properties");
+    Properties properties = new Properties();
+    properties.setProperty("pulse.oauth.provider", "uaa");
+    properties.setProperty("pulse.oauth.clientId", "pulse");
+    properties.setProperty("pulse.oauth.clientSecret", "secret");
+    // have the authorization uri point to a known uri that locator itself can serve
+    properties.setProperty("pulse.oauth.authorizationUri",
+        "http://localhost:" + locator.getHttpPort() + "/management");
+
+    properties.store(new FileWriter(pulsePropertyFile), null);
+    locator.startLocator();
+  }
+
+  @AfterClass
+  public static void cleanup() {
+    pulsePropertyFile.delete();
+  }
+
+  @Rule
+  public GeodeHttpClientRule client = new GeodeHttpClientRule(locator::getHttpPort);
+
+  @Test
+  public void testLogin() throws Exception {
+    HttpResponse response = client.get("/pulse/login.html");
+    // the request is redirect to the authorization uri configured before
+    assertResponse(response).hasStatusCode(200).hasResponseBody()
+        .containsOnlyOnce("latest")
+        .containsOnlyOnce("supported");
```

Comment:
Why only once? This looks fragile: If someone changes the page to show these words more than once, the test will fail, even if the request redirects properly.

Reply:
This is a random URL I picked up only for testing purpose. I happen to pick the /management uri, which returns json content {"latest":"...", "supported":[...]}. It is just a way to test the redirect works. It's not likely that url response would change and if it does, yes, developer will need to change the test.

Reply:
> Why only once? This looks fragile: If someone changes the page to show these words more than once, the test will fail, even if the request redirects properly.

From what I know that is highly unlikely. I think it is safe in this case.

Reply:
Each assertion should express a requirement that the test cares about. "Only once" is not a requirement here. Fix it.

Reply:
done.